src/mgr/MDSPerfMetricTypes.cc

```diff
@@ -0,0 +1,93 @@
+// -*- mode:C++; tab-width:8; c-basic-offset:2; indent-tabs-mode:t -*-
+// vim: ts=8 sw=2 smarttab
+
+#include <ostream>
+#include "mgr/MDSPerfMetricTypes.h"
+
+std::ostream& operator<<(std::ostream& os, const MDSPerfMetricSubKeyDescriptor &d) {
```

Comment:
your are replicating `OSDPerfMetricTypes.h` and `OSDPerfMetricTypes.cc`. is it possible to refactor `OSDPerfMetricTypes.*` instead, so we can reuse it?

Reply:
I tried that earlier -- templating the respective class, but couldn't get it fully working (probably needed more work related to templates). I can do that as a follow-up PR.